cmd/utils/utils.go

```diff
@@ -0,0 +1,47 @@
+package utils
+
+import (
+	"ldcli/cmd/cliflags"
+	"ldcli/internal/analytics"
+
+	"github.com/google/uuid"
+	"github.com/spf13/cobra"
+	"github.com/spf13/pflag"
+	"github.com/spf13/viper"
+)
+
+type CommandRunEventType struct {
+	EventName string
+}
+
+func (c CommandRunEventType) SendEvents(analyticsTracker analytics.Tracker, cmd *cobra.Command) {
+	analyticsTracker.SendEvent(
+		viper.GetString(cliflags.AccessTokenFlag),
+		viper.GetString(cliflags.BaseURIFlag),
+		"CLI Command Run",
+		c.buildProperties(cmd),
+	)
+}
+
+func (c CommandRunEventType) buildProperties(cmd *cobra.Command) map[string]interface{} {
+	id := uuid.New()
+	baseURI := viper.GetString(cliflags.BaseURIFlag)
+	var flags []string
+	if cmd.HasFlags() {
+		flagSet := cmd.Flags()
+		flagSet.Visit(func(f *pflag.Flag) {
+			flags = append(flags, f.Name)
+		})
+	}
```

Comment:
managed to dynamically get all the flags used for a command

Reply:
You could one-liner `cmd.Flags().Visit()` and get rid of the `if cmd.HasFlags()` check.